.github/workflows/ci.yml

```
@@ -0,0 +1,70 @@
+name: CI
+
+on: ['pull_request']
+
+jobs:
+  test:
+    name: 🧪 Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: 🔧 Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'yarn'
+          
+      - name: 📦 Install dependencies
+        run: yarn install --frozen-lockfile
+        
+      - name: 🧪 Run tests
+        run: yarn test:ci
+        
+      - name: 📊 Upload test results
+        if: always()
+        uses: actions/upload-artifact@v3
```

```suggestion
        uses: actions/upload-artifact@v4
```

This action is now on v4, and is much faster.